portality/crosswalks/article_crossref_xml.py

```diff
@@ -0,0 +1,304 @@
+
+        # volume
+
+        issue = journal.find("x:journal_issue", NS)
+
+        vol = issue.find("x:journal_volume", NS)
+        if vol is not None:
+            volume = _element(vol, "x:volume", NS)
+            if volume is not None:
+                bibjson.volume = volume
+
+        # issue
+        number = _element(issue, "x:issue", NS)
+        if number is not None:
+            bibjson.number = number
+
+        pages = record.find('x:pages', NS)
+        # start page
+        sp = _element(pages, "x:first_page", NS)
+        if sp is not None:
+            bibjson.start_page = sp
+
+        # end page
+        ep = _element(pages, "x:last_page", NS)
+        if ep is not None:
+            bibjson.end_page = ep
+
+        d = record.find("x:doi_data", NS)
+        if d is not None:
+            # doi
+            doi = _element(d, "x:doi", NS)
+            if doi is not None:
+                bibjson.add_identifier(bibjson.DOI, doi)
+
+            # fulltext
+            ftel = _element(d, "x:resource", NS)
+            if ftel is not None:
+                bibjson.add_url(ftel, "fulltext", NS)
+
+        ''''
+        # publisher record id
+        pri = _element(record, "publisherRecordId")
+        if pri is not None:
+            article.set_publisher_record_id(pri)
+
+        # document type
+        dt = _element(record, "documentType")
+        if dt is not None:
+            # FIXME: outstanding question as to what to do with this
+            pass
+        '''
+
+        # title
+        titles = record.find('x:titles', NS)
+        if titles is not None:
+            title = _element(titles, "x:title", NS)
+            if title is not None:
+                bibjson.title = title
+
+        # authors
+        contributors = record.find("x:contributors", NS)
+        contribs = contributors.findall("x:person_name", NS)
+        if contribs is not None:
+            for ctb in contribs:
+                if ctb.attrib["contributor_role"] == 'author':
+                    name = _element(ctb, "x:given_name", NS)
+                    name = name + ' ' + _element(ctb, "x:surname", NS)
+                    bibjson.add_author(name, affiliation=None)
+
+        # abstract
+        abstract_par = record.find("j:abstract", NS)
+        if abstract_par is not None:
+            text_elems = list(abstract_par.iter())
+            text = ""
+            if text_elems is not None:
+                for elems in text_elems:
+                    text = text + elems.text
+            bibjson.abstract = text[:30000]  # avoids Elasticsearch
+            # exceptions about .exact analyser not being able to handle
+            # more than 32766 UTF8 characters
+
+        '''
+        # keywords
+        keyel = record.find("keywords")
+        if keyel is not None:
+            for kel in keyel:
+                if kel.text != "":
+                    bibjson.add_keyword(kel.text)
+        '''
```

Comment:
Are these commented out sections hang-overs from the copy/paste from the DOAJ XML Crosswalk.  If so, can we remove them?  Or do we need to see if we can get this data from elsewhere in the Crossref XML?

Reply:
oh no, no, sorry, these may be removed. I overlooked that when cleaning.